### ISSUE TITLE
Reason for __import__ use instead of importlib ?

### ISSUE THREAD

**Issue by Carreau** [yt-project/yt#2297]
### Bug report

**Bug summary**

`__import__(..., level=-1)` [found here in `yt/analysis_modules/list_modules.py`](https://github.com/yt-project/yt/blob/master/yt/analysis_modules/list_modules.py#L36)raises `ValueError` on [weird use case](https://github.com/Carreau/frappuccino/issues/1). 

**Code for reproduction**

Kind of complicated... so let's not go this route of reproducing. I'm writing a tool that check what API has changed between two different package versions, and need to dynamically walk all reachable objects (stuff one do during scipy sprints).

The `level=-1` seem to create some issue and the docs of Python.org suggest to use importlib; is there a reason not to ? is `level=-1` necessary ? Just removing it allow my tool to happily generate change in API (e.g, between 3.4 and 3.5):

```diff
[snip]

- yt.visualization.fits_image.FITSImageData.writeto(self, fileobj, fields='None', clobber='False', **kwargs)
+ yt.visualization.fits_image.FITSImageData.writeto(self, fileobj, fields='None', overwrite='False', **kwargs)

- yt.visualization.fixed_resolution.FixedResolutionBuffer.export_fits(self, filename, fields='None', clobber='False', other_keys='None', units='cm')
+ yt.visualization.fixed_resolution.FixedResolutionBuffer.export_fits(self, filename, fields='None', overwrite='False', other_keys='None', units='cm', **kwargs)

[snip]
```

So I was just wondering if this `__import__` was normal.

cc @munkm with whom I found that this week-end.

**Comment by munkm** [comment]
Hi @Carreau ! So I checked on this and we've removed this code in our upcoming release (the yt 4.0 branch). I think `analysis_modules` is the only place where this form of a module import happens. I'll check and see if frappuccino works there!

**Comment by Carreau** [comment]
Works great on 4.0 !

**Comment by matthewturk** [comment]
Hooray!  Thanks for pinging about this.  I *think* I might have scraped that code from a flask extensions thing.

Also, frappuccino looks awesome!

**Comment by munkm** [comment]
Yes! Frappuccino is such a cool tool!! I'm looking forward to seeing what logo @Carreau comes up with for it too. ☕️ ❄️

**Comment by munkm** [comment]
Since this is fixed in 4.0 I'm going to close the issue, but feel free to reopen it if you think there's something else we could clarify!

**Comment by Carreau** [comment]
> Also, frappuccino looks awesome!

Thanks, let me know if you have any feature requests. 

> Yes! Frappuccino is such a cool tool!! I'm looking forward to seeing what logo @Carreau comes up with for it too. ☕️ ❄️

I'm gonna steal this emoji sequence. 

> Since this is fixed in 4.0 I'm going to close the issue, but feel free to reopen it if you think there's something else we could clarify!

Sounds good !
